src/routes/formsgSiteLaunch.ts

```diff
@@ -224,52 +222,118 @@ export class FormsgSiteLaunchRouter {
   ): Promise<void> => {
-    const subject = `[Isomer] Launch site ${repoName} domain validation`
-    let html = `<p>Isomer site ${repoName} is in the process of launching. (Form submission id [${submissionId}])</p>
-<p>Please set the following CNAME record:</p>
-<p>Source: ${domainValidationSource}</p>
-<p>Target: ${domainValidationTarget}</p>
-<p>Source: ${primaryDomainSource}</p>
-<p>Target: ${primaryDomainTarget}</p>`
-    if (redirectionDomain) {
-      html += `<p>Source: ${redirectionDomain.source}</p>\n<p>Target: ${redirectionDomain.target}</p>\n`
-    }
+    if (dnsRecordsEmailProps.length === 0) return
+    const { requesterEmail } = dnsRecordsEmailProps[0]
+    const subject = `[Isomer] DNS records for launching websites`
+
+    let html = `<p>Isomer sites are in the process of launching. (Form submission id [${submissionId}])</p>
+    <table>
+    <thead>
+      <tr>
+        <th>Repo Name</th>
+        <th>Source</th>
+        <th>Target</th>
+        <th>Type</th>
+      </tr>
+    </thead>
+    <tbody>`
+    dnsRecordsEmailProps.forEach((dnsRecords) => {
+      // check if dnsRecords.redirectionDomain is undefined
+      const isRedirection = !!dnsRecords.redirectionDomain
+      html += `
+    <tr>
+      <td>${dnsRecords.repoName}</td>
+      <td>${dnsRecords.domainValidationSource}</td>
+      <td>${dnsRecords.domainValidationTarget}</td>
+      <td>CNAME</td>
+    </tr>
+    <tr>
+      <td>${dnsRecords.repoName}</td>
+      <td>${
+        isRedirection
+          ? // if redirection, website will be hosted in the 'www' subdomain
+            `www.${dnsRecords.primaryDomainSource}`
+          : dnsRecords.primaryDomainSource
+      }</td>
+      <td>${dnsRecords.primaryDomainTarget}</td>
+      <td>CNAME</td>
+    </tr>`
+
+      if (isRedirection) {
+        html += `
+      <tr>
+        <td>${dnsRecords.repoName}</td>
+        <td>${dnsRecords.redirectionDomain?.source}</td>
+        <td>${dnsRecords.redirectionDomain?.target}</td>
+        <td>A Record</td>
+      </tr>`
+      }
+    })
+    html += `</tbody></table>
+    <p>This email was sent from the Isomer CMS backend.</p>`
+    await mailer.sendMail(requesterEmail, subject, html)
+  }
 
-    html += `<p>This email was sent from the Isomer CMS backend.</p>`
-    await mailer.sendMail(requestorEmail, subject, html)
+  private async siteLaunchHandler(
+    formResponses: FormResponsesProps[],
+    submissionId: string
+  ) {
+    const launchResults = await Promise.all(formResponses.map(this.siteLaunch))
+    const successResults: DnsRecordsEmailProps[] = []
+    launchResults.forEach((launchResult) => {
+      if (launchResult.isOk()) {
+        successResults.push(launchResult.value)
+      }
+    })
+
+    await this.sendDNSRecords(submissionId, successResults)
+
+    const failureResults: LaunchFailureEmailProps[] = []
+
+    launchResults.forEach((launchResult) => {
+      if (launchResult.isErr() && launchResult.error) {
```

Comment:
will there be any errors that don't contain the `error` property? they'll be silently ignored

Reply:
The method call to `siteLaunch` always returns the error with the form responses. But you are right since this is not a guarantee if `siteLaunch()` changes. Have defensively added a failure statement here: 1b19f52e